--- a/CMake/OtherTests.cmake
+++ b/CMake/OtherTests.cmake
@@ -226,39 +226,6 @@ if(NOT DEFINED CMAKE_TOOLCHAIN_FILE)
   elseif(HAVE_POLL_H)
     set(CMAKE_REQUIRED_FLAGS "-DHAVE_POLL_H")
   endif()
-  check_c_source_runs("
-    #include <stdlib.h>
-    #include <sys/time.h>
-
-    #ifdef HAVE_SYS_POLL_H
-    #  include <sys/poll.h>
-    #elif  HAVE_POLL_H
-    #  include <poll.h>
-    #endif
-
-    int main(void)
-    {
-        if(0 != poll(0, 0, 10)) {
-          return 1; /* fail */
-        }
-        else {
-          /* detect the 10.12 poll() breakage */
-          struct timeval before, after;
-          int rc;
-          size_t us;
-
-          gettimeofday(&before, NULL);
-          rc = poll(NULL, 0, 500);
-          gettimeofday(&after, NULL);
-
-          us = (after.tv_sec - before.tv_sec) * 1000000 +
-            (after.tv_usec - before.tv_usec);
-
-          if(us < 400000) {
-            return 1;
-          }
-        }
-        return 0;
-    }" HAVE_POLL_FINE)
+  set(HAVE_POLL_FINE 1)
 endif()
 
